apps/sv/frontend/src/components/governance/ProposalVoteForm.tsx

```diff
@@ -0,0 +1,230 @@
+        Your Vote
+      </Typography>
+
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          e.stopPropagation();
+          form.handleSubmit();
+        }}
+      >
+        <Box sx={{ display: 'flex', flexDirection: 'column', gap: 2 }}>
+          <form.Field
+            name="url"
+            validators={{
+              onChange: ({ value }) => {
+                const result = z
+                  .string()
+                  .optional()
+                  // URL is optional so we allow undefined or empty string here as it's the default value
+                  .refine(url => !url || url.trim() === '' || isValidUrl(url), {
+                    message: 'Invalid URL',
+                  })
+                  .safeParse(value);
+                return result.success ? undefined : result.error.issues[0].message;
+              },
+            }}
+            children={field => {
+              return (
+                <TextField
+                  label="URL"
+                  name={field.name}
+                  value={field.state.value}
+                  onBlur={field.handleBlur}
+                  onChange={e => field.handleChange(e.target.value)}
+                  error={!field.state.meta.isValid}
+                  helperText={
+                    <span data-testid="your-vote-url-helper-text">
+                      {field.state.meta.errors?.[0]}
+                    </span>
+                  }
+                  inputProps={{ 'data-testid': 'your-vote-url-input' }}
+                />
+              );
+            }}
+          />
+          <form.Field
+            name="reason"
+            validators={{
+              onChange: ({ value }) => {
+                const result = z.string().safeParse(value);
+                return result.success ? undefined : result.error.issues[0].message;
+              },
+            }}
+            children={field => {
+              return (
+                <TextField
+                  label="Reason"
+                  multiline
+                  rows={4}
+                  name={field.name}
+                  value={field.state.value}
+                  onBlur={field.handleBlur}
+                  onChange={e => field.handleChange(e.target.value)}
+                  error={!field.state.meta.isValid}
+                  helperText={field.state.meta.errors?.[0]}
+                  inputProps={{ 'data-testid': 'your-vote-reason-input' }}
+                />
+              );
+            }}
+          />
+        </Box>
+
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 4,
+            justifyContent: 'center',
+            alignItems: 'center',
+            mt: 4,
+          }}
+        >
+          <form.Field
+            name="vote"
+            validators={{
+              // Not using onChange here because by default, no vote is selected
+              onMount: ({ value }) => {
```

Comment:
Normally, we only validate after a change has been made(`onChange`) but in this case, we want the form to be invalid until a vote is selected. As a result, the vote is validated `onMount`. No error is shown on the screen though as i'm hoping it's obvious you have to actually select a vote. See screenshots